main.go

```diff
@@ -85,17 +89,22 @@ func main() {
 
 	metrics.Register()
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), options)
+	kubeConfig := ctrl.GetConfigOrDie()
+	if kubeConfig.UserAgent == "" {
```

Comment:
would this typically be populated? by who?

Reply:
it could come from the kubeConfig file. In a k8s environment, the default kube-config doesn't have an agent.